nixos/modules/services/admin/realmd.nix

```diff
@@ -0,0 +1,49 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  pkg = pkgs.realmd;
+  cfg = config.services.realmd;
+in
+{
+  options.services.realmd = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable the realmd service which allows using `realm` to join
+        machines to AD/LDAP domains.
+      '';
+    };
+
+    configText = mkOption {
+      type = types.lines;
+      default = "";
+      description = "The verbatim contents of config file /etc/realmd.conf";
+    };
```

Comment:
I recommend you to use [RFC0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) to have structural settings instead of stringly config text:

```diff
   pkg = pkgs.realmd;
   cfg = config.services.realmd;
+
+  settingsFormat = pkgs.formats.ini { };
+  configFile = settingsFormat.generate "realmd.conf" cfg.settings;
 in
 {
   options.services.realmd = {
@@ -18,8 +24,14 @@ in
     };
 
     configText = mkOption {
-      type = types.lines;
-      default = "";
+      type = lib.types.submodule { freeformType = settingsFormat.type; };
+      default = { };
+      example = {
+        users = {
+          default-home = "/home/%U";
+          default-shell = "${pkgs.bash}/bin/bash";
+        };
+      };
       description = "The verbatim contents of config file /etc/realmd.conf";
     };
   };
@@ -41,9 +53,6 @@ in
       };
     };
 
-    environment.etc."realmd.conf" = {
-      target = "realmd.conf";
-      source = pkgs.writeText "realmd.conf" "${cfg.configText}";
-    };
+    environment.etc."realmd.conf".source = configFile;
   };
 }
```

What do you think?